server/routes/api.js

```diff
@@ -0,0 +1,87 @@
+const express = require("express");
+const router = express.Router();
+const moment = require("moment");
+const Expense = require("../models/Expense");
+const mongoose = require("mongoose");
+const Schema = mongoose.Schema;
+
+router.get("/expenses", function(req, res) {
+  if (req.query.d1) {
+    let d1 = moment(req.query.d1).format("LLLL");
+    let d2 = req.query.d2
+      ? moment(req.query.d2).format("LLLL")
+      : moment().format("LLLL");
+    let findProm = Expense.find({
+      $and: [{ date: { $gt: d1 } }, { date: { $lt: d2 } }]
+    }).sort({date : -1});
+    findProm.then(function(result) {
+      res.send(result);
+    });
+  } else {
+    Expense.find({})
+      .sort({ date: -1 })
+      .exec(function(err, result) {
+        res.send(result);
+      });
+  }
+});
+
+router.get("/expenses/:group", function(req, res) {
+  let group = req.params.group;
+  let total = req.query.total;
+  if (total === "true") {
+    let query = Expense.aggregate([
+      { $match: { group: group } },
+      {
+        $group: {
+          _id: "$group",
+          TotalAmount: { $sum: "$amount" }
+        }
+      }
+    ]);
+    query.then(function(result) {
+      res.send(result);
+    });
+  } else {
+    let query = Expense.find({ group: group });
+    query.then(function(result) {
+      res.send(result);
+    });
+  }
+});
+
+router.post("/new", function(req, res) {
+  const data = req.body;
+  if (data.name && data.amount && data.group) {
+    const item = new Expense({
+      item: data.name,
+      amount: data.amount,
+      group: data.group,
+      date: data.date
+        ? moment(data.date).format("LLLL")
```

Comment:
This is a duplicate code. You have the same code somewhere else, use one function to do that